### DISCUSSION TITLE
Using wildfly-clustering-spring-session with the Wildfly embedded Infinispan subsystem

### DISCUSSION THREAD

**Discussion by InfoSec812** [wildfly-clustering/wildfly-clustering-spring-session#104] [Q&A]
It appears that the documentation for this library is aimed at using an external Infinispan cluster and I am wondering if it can be configured to work with the Infinispan subsystem in JBoss Wildfly instead?

Does Wildfly's Infinispan subsystem export the port 11222 for the HotRod protocol, or is there another way to access the cache manager?

Thanks in advance!

**Comment by pferraro** [comment]
The README documents both types of session managers (search for "@EnableInfinispanHttpSession" or "@EnableInfinispanIndexedHttpSession"), though admittedly this should be improved, since all of the examples reference the HotRod-based session manager.

To setup the remote Infinispan cluster for use with the HotRod-based session manager, you would use Infinispan Server [1], not WildFly.

The embedded Infinispan-based session manager is not currently equipped for use with an injected cache configuration (e.g. injected from WildFly's infinispan subsystem).

That said, if your target platform is WildFly or one of its derivatives, it is unclear to me why you would prefer to use this project rather than WildFly's native distributed session manager.  The underlying implementation is identical.  This project merely adapts the WildFly clustering's session manager for use with the Spring Session API rather than the API from Undertow (the servlet container used by WildFly).  Perhaps it might make sense if an application depends on a session-dependent authentication mechanism from Spring Security and thus requires Spring Session.  Otherwise, you are much better off using WildFly's distributed session manager and avoiding all of the caveats enumerated here: https://github.com/wildfly-clustering/wildfly-clustering-spring-session#notes

Let me know if you have any further questions.

[1] https://infinispan.org/download/
[1] https://infinispan.org/docs/stable/titles/server/server.html
